appeals/web/src/server/appeals/personal-list/personal-list.mapper.js

```diff
@@ -5,27 +5,93 @@ import { preRenderPageComponents } from '#lib/nunjucks-template-builders/page-co
 import { dateToDisplayDate } from '#lib/dates.js';
 import { numberToAccessibleDigitLabel } from '#lib/accessibility.js';
 import * as authSession from '../../app/auth/auth-session.service.js';
+import { appealStatusToStatusTag } from '#lib/nunjucks-filters/status-tag.js';
+import { capitalize } from '#lib/string-utilities.js';
 
 /** @typedef {import('@pins/appeals').AppealList} AppealList */
 /** @typedef {import('@pins/appeals').Pagination} Pagination */
 /** @typedef {import('../../app/auth/auth.service').AccountInfo} AccountInfo */
 
 /**
  * @param {AppealList|void} appealsAssignedToCurrentUser
+ * @param {string} urlWithoutQuery
+ * @param {string|undefined} appealStatusFilter
  * @param {import("express-session").Session & Partial<import("express-session").SessionData>} session
  * @returns {PageContent}
  */
-export function personalListPage(appealsAssignedToCurrentUser, session) {
+
+export function personalListPage(
+	appealsAssignedToCurrentUser,
+	urlWithoutQuery,
+	appealStatusFilter,
+	session
+) {
 	const account = /** @type {AccountInfo} */ (authSession.getAccount(session));
 	const userGroups = account?.idTokenClaims?.groups ?? [];
 	const isInspector = userGroups.includes(config.referenceData.appeals.inspectorGroupId);
 
+	const filterItemsArr = ['all', ...(appealsAssignedToCurrentUser?.statuses || [])].map(
+		(appealStatus) => ({
+			text: capitalize(appealStatusToStatusTag(appealStatus)),
+			value: appealStatus,
+			selected: appealStatusFilter === appealStatus
+		})
+	);
+
 	/** @type {PageContent} */
 	const pageContent = {
 		title: 'Personal list',
 		heading: 'Cases assigned to you',
 		headingClasses: 'govuk-heading-l govuk-!-margin-bottom-6',
 		pageComponents: [
+			{
+				type: 'details',
+				parameters: {
+					summaryText: 'Filters',
+					html: '',
+					pageComponents: [
+						{
+							type: 'html',
+							parameters: {
+								html: `<form method="GET">`
```

Comment:
This is totally fine, nothing wrong with this approach, but just so you're aware, the `PageComponent` type accepts a `wrapperHtml` parameter allowing you to define opening and closing html, to avoid the need for a separate component

Reply:
I'm also working on simplifying the page components system to remove PageComponentGroup, but that shouldn't affect this

Reply:
on using html vs wrapperHtml - i think it's more clear to see the tree structure with this